src/test/java/uk/gov/cshr/controller/signup/SignupControllerTest.java

```diff
@@ -219,7 +248,8 @@ public void shouldRedirectToEnterTokenPageIfInviteNotAuthorised() throws Excepti
         Invite invite = new Invite();
         invite.setAuthorisedInvite(false);
 
-        when(inviteService.isInviteValid(code)).thenReturn(true);
+        when(inviteService.isCodeExists(code)).thenReturn(true);
+        when(inviteService.isCodeExpired(code)).thenReturn(false);
```

Comment:
Why the isInviteValid(code) method replaced with the isCodeExists(code) and isCodeExpired(code)?
Is isInviteValid(code) returns the different result?

Reply:
The isInviteValid(code) method is combination of both codeExists and codeExpired. Because of the new functionality need to check both existence and expiry separately and hence added these new method instead of changing the existing one as existing was already referred few places.

Reply:
If a new code need to be tested then I suggest that please add a nw unit test method if it is easy to do so.

Reply:
The logic within the code is changed and isinviteValid is no longer used and hence removed. The new added method are used in the new logic (to implement new workflow) and hence there is no option but to change as above otherwise test would not make any sense.